### ISSUE TITLE
pydemo bug

### ISSUE THREAD

**Issue by ysyecust** [hugeproblem/nged#1]
I used pydemo with this case 
![image](https://github.com/hugeproblem/nged/assets/34128364/2151720e-35df-4490-8727-9f3fac87234f)
but when I add code "return fetch(0)+fetch(1)" in the code block, the program crash. My system is windows11 python 3.11.5

**Comment by hugeproblem** [comment]
![image](https://github.com/hugeproblem/nged/assets/100833277/cdc3764b-d508-4e2c-aa5e-2dee34181f26)

Sorry but failed to reproduce your problem, could you provide more details about the list of python packages you've installed?

And please copy the nodes and links, then paste the content here

**Comment by hugeproblem** [comment]
![image](https://github.com/hugeproblem/nged/assets/100833277/1a235653-7ffd-4f4e-b53a-8ee746a122c0)

tested on Windows 11 + Python 3.11.5, still failed to reproduce the problem

the test graph is:

```json
{"items":[{"aabb":{"max":{"x":30.0,"y":15.0},"min":{"x":-30.0,"y":-15.0}},"color":"#ffffffff","f":"node","id":6603431362,"name":"np.arange","parms":{"range":[0.0,10.0],"step":1.0},"pos":[-93,-101],"pynode":"","type":"np.arange","uid":"b1c3a740-fc51-440c-947f-4bbf2b108528"},{"aabb":{"max":{"x":25.0,"y":25.0},"min":{"x":-25.0,"y":-25.0}},"color":"#689f38ff","f":"node","id":1472294687,"name":"Output","parms":{},"pos":[2,141],"pynode":"","type":"output","uid":"7bc744ba-ac1a-4b9f-b6e3-d69f0a759cc5"},{"aabb":{"max":{"x":25.0,"y":20.0},"min":{"x":-25.0,"y":-20.0}},"color":"#81d4faff","f":"node","id":10515940801,"name":"code","parms":{"code":"import platform\nimport sys\nimport numpy\nreturn (fetch(0)+fetch(1),\n        'PY: '+sys.version,\n        'NP: '+numpy.__version__,\n        'OS: '+platform.platform())","ui_only":null},"pos":[2,32],"pynode":"","type":"code_snippet","uid":"6d3bfc0c-ccb7-4367-8fcd-876129568c8d"},{"aabb":{"max":{"x":30.0,"y":15.0},"min":{"x":-30.0,"y":-15.0}},"color":"#ffffffff","f":"node","id":13926435273,"name":"np.arange","parms":{"range":[0.0,10.0],"step":1.0},"pos":[93,-96],"pynode":"","type":"np.arange","uid":"cae03582-328c-4c3b-a196-5f27289a3f30"}],"links":[{"from":{"id":13926435273,"port":0},"to":{"id":10515940801,"port":1}},{"from":{"id":10515940801,"port":0},"to":{"id":1472294687,"port":0}},{"from":{"id":6603431362,"port":0},"to":{"id":10515940801,"port":0}}]}
```

// _above content can be directly pasted into a node graph_ &uArr;

Please proved more details on how to reproduce the problem

**Comment by hugeproblem** [comment]
In your example there do have a bug where `fetch(1)` should return None but returned the result of second input node instead, this has been fixed in [commit #fb10c78c7418e41f8acac8ac80e192896bf051e9](https://github.com/hugeproblem/nged/commit/fb10c78c7418e41f8acac8ac80e192896bf051e9)

Hope this commit also fixed your problem, since there is no consist way of reproducing your problem, I'm closing this issue.
